In [47]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

In [48]:
tweet = pd.read_csv('Data/train.csv')

In [49]:
tweet.isnull().sum()

id             0
keyword       61
location    2533
text           0
target         0
dtype: int64

**keyword y location** tienen valores nulos

Para este analisis solo nos quedamos con los campos **text y target**

In [50]:
tweet = tweet[['text','target']]

Agregamos los campos.

**logitud_tweet**: Cantidad de caracteres del tweet

**cant_palabras**: Cantidad de palabras del tweet

**long_min_palabras**: Es la longitud minima de la palabra mas chica del tweet

**long_max_palabras**: Es la longitud maxima de la palabra mas grande del tweet

**long_prom_palabras**: Es la longitud promedio de las palabras del tweet

In [52]:
tweet['longitud_tweet'] = tweet['text'].str.len()
tweet['cant_palabras'] = tweet['text'].str.split().map(lambda x: len(x))
tweet['long_min_palabras'] = tweet['text'].str.split().apply(lambda x : [len(i) for i in x]).map(lambda x: np.min(x)).astype(int)
tweet['long_max_palabras'] = tweet['text'].str.split().apply(lambda x : [len(i) for i in x]).map(lambda x: np.max(x)).astype(int)
tweet['long_prom_palabras'] = tweet['text'].str.split().apply(lambda x : [len(i) for i in x]).map(lambda x: np.mean(x))

In [110]:
tweet['long_max_palabras'].value_counts(normalize=True)

22    0.458295
9     0.063838
10    0.061999
23    0.058190
11    0.055563
12    0.051228
8     0.044792
13    0.041377
14    0.034283
15    0.028898
7     0.028110
16    0.027716
6     0.011822
5     0.006174
17    0.005123
18    0.003547
20    0.002627
19    0.002627
24    0.001839
25    0.001708
28    0.001314
21    0.001314
4     0.001314
26    0.001314
27    0.001051
29    0.000657
31    0.000657
30    0.000525
34    0.000394
41    0.000263
37    0.000131
50    0.000131
3     0.000131
62    0.000131
56    0.000131
42    0.000131
40    0.000131
33    0.000131
35    0.000131
57    0.000131
43    0.000131
Name: long_max_palabras, dtype: float64

Tenemos 3489 tweet con una longitud maxima de palabra 22 el cual representa casi un 46% de los tweet totales lo cual resulta alto si consideramos que solo estamos tomando un solo atrinuto de los tweet en este caso las palabras de longitud maxima igual a 22.

Si miramos estos tweet podemos ver que contienen una url lo cual resulta sospechoso